src/components/contentItem/ContentItem.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react';
+
+import { CURRENCY } from '@constants/constants';
+import plusOrange from '@assets/plusOrange.svg';
+import { addPizza } from '@store/pizzaSlice';
+import { ContentItemType, SizesPizzaType } from '@ts/types/types';
+import PizzaSize from './pizzaSize/PizzaSize';
+import PizzaType from './pizzaType/PizzaType';
+import { useAppDispatch } from '../../hook';
+
+import './ContentItem.scss';
+
+export default function ContentItem({ pizza, order }: ContentItemType) {
+  const typeOfPizza: string = Object.keys(pizza.price)[0];
+  const [type, setType] = useState<string>(typeOfPizza);
+  const [size, setSize] = useState<26 | 30 | 40>(() => {
+    const firstPizzaType = Object.keys(
+      pizza.price
+    )[0] as keyof typeof pizza.price;
+    return Object.keys(
+      pizza.price[firstPizzaType] || ({} as Record<SizesPizzaType, number>)
+    )[0] as unknown as SizesPizzaType;
+  });
+
+  const [price, setPrice] = useState(0);
+  const count = order.find((e) => e.pizza.id === pizza.id)?.params?.count || 0;
+  const dispatch = useAppDispatch();
```

Comment:
Я бы сделал что-то такое

```suggestion
  const [type, setType] = useState<string>(); // ну и тут не string же
  const [size, setSize] = useState<26 | 30 | 40>();
  const [price, setPrice] = useState();
  
  const dispatch = useAppDispatch();
  
  const count = order.find((e) => e.pizza.id === pizza.id)?.params?.count || 0;
  
  useEffect(() = {
    const typeOfPizza: string = Object.keys(pizza.price)[0]; 
    setType(typeOfPizza);
    
    const firstPizzaType = Object.keys(pizza.price)[0] as keyof typeof pizza.price;
    const pizzaSize = Object.keys(pizza.price[firstPizzaType] || ({} as Record<SizesPizzaType, number>))[0] as unknown as SizesPizzaType; // очень странная конструкция, ищи как упрощать. В прод такое не пропустят
    setSize(pizzaSize);
  }, [])
```